packages/commonwealth/client/scripts/state/api/proposals/fetchActiveCosmosProposals.ts

```diff
@@ -2,9 +2,11 @@ import app from 'state';
 import { useQuery } from '@tanstack/react-query';
 import Cosmos from 'controllers/chain/cosmos/adapter';
 import { getActiveProposals } from 'controllers/chain/cosmos/gov/utils';
+import { CosmosProposal } from 'controllers/chain/cosmos/gov/v1beta1/proposal-v1beta1';
 
-const fetchActiveProposals = async () => {
-  return getActiveProposals(app.chain as Cosmos);
+const fetchActiveProposals = async (): Promise<CosmosProposal[]> => {
+  if (!(app.chain instanceof Cosmos)) return Promise.resolve([]);
```

Comment:
Rather than checking the chain instance here consider using [this method](https://github.com/hicommonwealth/commonwealth/pull/4769/files#diff-4d7ae894407de1ecbbfcefde5c960da2281f2f15b057e64f3d49e86002e13c5eR56). Basically, pass `app.chain?.network === ChainNetwork.Aave` when you define `isApiReady` in your `useCompletedCosmosProposalsQuery` calls.

Reply:
See: https://tanstack.com/query/v4/docs/react/guides/dependent-queries

>The projects query will start in:

```
status: 'loading'
fetchStatus: 'idle'
```

Reply:
how about this 

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/1f4ad34d-1360-4ec8-8b27-85c933cc9204)

Reply:
https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#the-idle-state-has-been-removed

Reply:
Good find. I will play around with this.

Reply:
The 'fetching' status is nice to have, but I found that I didn't need it after all. the best UX is using the default RQ isLoading, combined with a check for `isCosmos` to show the spinners.

So now, @timolegros I am using your suggestion, but still checking within the `fetchActiveCosmosProposals` and `fetchCompletedCosmosProposals` themselves (instead of proposals.ts), because it applies to all uses of these queries.